core/src/main/java/tachyon/thrift/ClientFileInfo.java

```diff
@@ -610,7 +610,7 @@ public void addToBlockIds(long elem) {
   }
 
   public List<Long> getBlockIds() {
-    return this.blockIds;
+    return this.blockIds == null ? new ArrayList<Long>() : this.blockIds;
```

Comment:
So changed thrift proto to let thrift gen to generate empty collection instead of null.